samples/bluetooth/direct_test_mode/README.rst

```diff
@@ -0,0 +1,237 @@
+
+The DTM application contains a driver for a 2-wire UART interface.
+The driver maps two-octet commands and events to the DTM library, as specified by the Bluetooth Low Energy DTM specification.
+
+.. figure:: /images/bt_dtm_dut.svg
+   :alt: nRF5 with DTM as a DUT
+
+The implementation is self-contained and requires no Bluetooth Low Energy protocol stack for its operation.
+The MPU is initialized in the standard way.
+The DTM library function ``dtm_init`` configures all interrupts, timers, and the radio.
+
+``main.c`` may be replaced with other interface implementations, such as an HCI interface, USB, or another interface required by the Upper Tester.
+
+The interface to the Lower Tester uses the antenna connector of the chosen nRF5 Development Kit.
+While in principle an aerial connection might be used, conformance tests cover the reading of the transmission power delivered by the DUT.
+For this reason, a coaxial connection between the DUT and the Lower Tester is employed for all conformance testing.
+
+DTM module interface
+====================
+
+The DTM function ``dtm_cmd_put`` implements the four commands defined by the Bluetooth Low Energy standard:
+
+* TEST SETUP (called RESET in Bluetooth 4.0)
+* RECEIVER_TEST
+* TRANSMITTER_TEST
+* TEST_END
+
+In the ``dtm_cmd_put`` interface, DTM commands are accepted in the 2-byte format.
+Parameters such as: CMD code, Frequency, Length or Packet Type are encoded within this command.
+
+The following DTM events are polled using the ``dtm_event_get`` function:
+
+* PACKET_REPORTING_EVENT
+* TEST_STATUS_EVENT [SUCCESS|FAIL]
+
+.. figure:: /images/bt_dtm_engine.svg
+   :alt: State machine overview of the DTM
+
+Supported PHYs
+==============
+
+The DTM application supports all four PHYs specified in DTM, but not all devices support all the PHYs.
+
+.. list-table:: Supported PHYs
+   :header-rows: 1
+
+   * - PHY
+     - nRF5340
+   * - LE 1M
+     - Yes
+   * - LE 2M
+     - Yes
+   * - LE Coded S=8
+     - Yes
+   * - LE Coded S=2
+     - Yes
+
+Vendor Specific packet payload
+==============================
+
+The Bluetooth Low Energy 2-wire UART DTM interface standard reserves the Packet Type, also called payload parameter, with binary value '11' for a Vendor Specific packet payload.
+
+The DTM command is interpreted as a Vendor Specific one when both the following conditions are met:
+
+* Its CMD field is set to Transmitter Test, binary '10'.
+* Its PKT field is set to Vendor Specific, binary '11'.
+
+Vendor specific commands can be divided into different categories as follows:
+
+* If the Length field is set to 0 (symbol ``CARRIER_TEST``), an unmodulated carrier is turned on at the channel indicated by Frequency field.
+  It remains turned on until a ``TEST_END`` or ``RESET`` command is issued.
+* If the Length field is set to 1 (symbol ``CARRIER_TEST_STUDIO``), is used by the nRFgo Studio to indicate that an unmodulated carrier is turned on at the channel.
```

Comment:
This command is here to maintain backward compatibility with DTM application from nRF5 SDK. I didn't mention it in the requirements, because it is not necessary to use this command to have DTM working.